### ISSUE TITLE
Add option to allow use of platform-linefeed (`YAMLGenerator.Feature.USE_PLATFORM_LINE_BREAKS`)

### ISSUE THREAD

**Issue by thauk-copperleaf** [FasterXML/jackson-dataformats-text#84]
The behaviour I expect would be for the serializer to produce output with line endings consistent with that platform it's running on, unless explicitly overriden.

I'm just wondering if this is on purpose or not?

Deeper dive:

`YAMLGenerator` uses `DumperOptions` from snakeyaml and overrides some settings, but does not override the line ending setting:

https://github.com/FasterXML/jackson-dataformats-text/blob/a0f5c80bb431dbba9a9f505c3f1554aa4be0bfc5/yaml/src/main/java/com/fasterxml/jackson/dataformat/yaml/YAMLGenerator.java#L231-L255

And since `DumperOptions` has the following default:
```    private LineBreak lineBreak = LineBreak.UNIX;```

That results in output having Unix line endings, regardless of platform.

Alternatively, can you tell me how to produce YAML with platform-specific line endings?

**Comment by cowtowncoder** [comment]
Generator uses SnakeYAML defaults so this is not conscious definition. I actually do not know what YAML specs says: I assume it probably allows multiple styles of linefeeds just like JSON and XML.
Whether platform defaults should be used or canonical defaults is an open question; since both are (I presume) legit I don't have strong opinion.

I would be open to having an option to change behavior: being an API change, it needs to go in Jackson 3.0 however.

One challenge there is this: it is easy to add on/off kinds of things (`YAMLGenerator.Feature`).
However, with 3.0 it will be much easier to add backend-specific configuration via Builder-style construction. But to keep dependencies to `SnakeYAML` hidden, we'd probably need a new enumeration for choices as well.
That seems cleaner than adding logic to detect platform, although maybe that'd be fine as an alternative. But that would then require definition of some kind of universal default, for feature for "enable platform-specific linefeed" (if disabled, what is the linefeed then -> default).

**Comment by thauk-copperleaf** [comment]
@cowtowncoder 

Thanks for the quick reply!

> I actually do not know what YAML specs says: I assume it probably allows multiple styles of linefeeds just like JSON and XML.

Correct. The [YAML spec](http://yaml.org/spec/1.1/#id871136) allows four different "generic line breaks", including CRLF for Windows and LF for Unix.

> Whether platform defaults should be used or canonical defaults is an open question; since both are (I presume) legit I don't have strong opinion.

The behaviour I expect from the serializer is that it outputs textual data with line endings consistent with that platform it's running on, by default, but should allow overriding to a specific line ending. I think this is the least-surprising choice (between that and defaulting to LF), and indeed, is the design choice that's already present because of https://github.com/FasterXML/jackson-databind/issues/585 so at least making that change would be internally consistent.

> One challenge there is this: it is easy to add on/off kinds of things (YAMLGenerator.Feature)

Would you accept a PR to add a new enumeration value to `YAMLGenerator.Format` called `USE_PLATFORM_LINE_BREAKS` and adds the following to `YAMLGenerator#buildDumperOptions`?

```
        if (Feature.USE_PLATFORM_LINE_BREAKS.enabledIn(_formatFeatures)) {
            opt.setLineBreak(opt.LineBreak.getPlatformLineBreak());
        }
```

To not break existing clients, the default value of `USE_PLATFORM_LINE_BREAKS` would be set to `false`

**Comment by cowtowncoder** [comment]
Forgot to close, was included in 2.9.6.
